tools/buildgen/extract_metadata_from_bazel_xml.py

```diff
@@ -293,8 +314,9 @@ def _expand_intermediate_deps(target_dict, public_dep_names, bazel_rules):
         elif external_dep_name_maybe:
             deps.add(external_dep_name_maybe)
 
-        elif bazel_dep.startswith(
-                '//external:') or not bazel_dep.startswith('//'):
+        elif bazel_dep.startswith('//external:') or (
+                not bazel_dep.startswith('//') and
+                not RE_SUPPORTED_LIB.match(bazel_dep)):
```

Comment:
In this context, the name "RE_SUPPORTED_LIB" gives you no useful insight into what is it you're attempting to do.

I think I kind of understand the point of what you're trying to do (you are trying to make sure that the .proto files you want actually do get expanded into the list of your source files), but that doesn't really fit the function name "_expand_intermediate_deps", it's description and also https://github.com/grpc/grpc/blob/026046d84301999936a759c1779669c04c6ffaaa/tools/buildgen/extract_metadata_from_bazel_xml.py#L351 

My first thought is that perhaps instead of trying to hack the .proto extraction logic into the existing _expand_intermediate_deps function, you could just introduce a separate function (e.g "_expand_proto_files_from_external_deps") that simply looks for the .proto files we want and adds them where they belong. That's probably a few lines of code and makes the intent clearer.